showcase/gapic-showcase/src/test/java/com/google/showcase/v1beta1/it/ITDynamicRoutingHeaders.java

```diff
@@ -0,0 +1,308 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.showcase.v1beta1.it;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.api.gax.httpjson.ApiMethodDescriptor;
+import com.google.api.gax.httpjson.ForwardingHttpJsonClientCall;
+import com.google.api.gax.httpjson.ForwardingHttpJsonClientCallListener;
+import com.google.api.gax.httpjson.HttpJsonCallOptions;
+import com.google.api.gax.httpjson.HttpJsonChannel;
+import com.google.api.gax.httpjson.HttpJsonClientCall;
+import com.google.api.gax.httpjson.HttpJsonClientInterceptor;
+import com.google.api.gax.httpjson.HttpJsonMetadata;
+import com.google.common.collect.ImmutableList;
+import com.google.showcase.v1beta1.EchoClient;
+import com.google.showcase.v1beta1.EchoRequest;
+import com.google.showcase.v1beta1.it.util.TestClientInitializer;
+import io.grpc.CallOptions;
+import io.grpc.Channel;
+import io.grpc.ClientCall;
+import io.grpc.ClientInterceptor;
+import io.grpc.ForwardingClientCall;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ITDynamicRoutingHeaders {
```

Comment:
I'll look to add explicit cases for multiple matches. There is some differences between the echo rpc and the routing proto test cases, but I'll try to add as much as I can.

Reply:
Sounds good thanks!

Reply:
If there is any test case that we think is missing from the showcase protos, we should definitely add them to the protos in `gapic-showcase`. It could be a separate PR also if the effort is large.